main.js

```diff
@@ -0,0 +1,109 @@
+var products = [];
+var seasonalSales = [];
+const discountSelect = document.getElementById("discountSelect");
+const productDisplayDiv = document.getElementById("productDisplayDiv");
+productDisplayDiv.innerHTML += `<div class="col-md-4"><div class="panel panel-primary"><div class="panel-heading"><h2 class="panel-title">Apparel</h2></div><div class="panel-body" id="apparel"></div></div></div>`;
+productDisplayDiv.innerHTML += `<div class="col-md-4"><div class="panel panel-primary"><div class="panel-heading"><h2 class="panel-title">Furniture</h2></div><div class="panel-body" id="furniture"></div></div></div>`;
+productDisplayDiv.innerHTML += `<div class="col-md-4"><div class="panel panel-primary"><div class="panel-heading"><h2 class="panel-title">Household</h2></div><div class="panel-body" id="household"></div></div></div>`;
+const apparelDiv = document.getElementById("apparel");
+const furnitureDiv = document.getElementById("furniture");
+const householdDiv = document.getElementById("household");
+var discountApplied = { //flags to determine whether discounts have already been triggered or not
+    "Autumn": false,
+    "Winter": false,
+    "Spring": false
+};
+
+discountSelect.addEventListener("change", discountPrices);
+
+function discountPrices(e) {
+    let selectedSeason = (e.target.value);
+    if (discountApplied[selectedSeason] === false) { //only discounts each price one time
+        let selectedDiscount = getDiscount(selectedSeason);
+        let selectedCategory = getCategory(selectedSeason);
+        let selectedDiv = document.getElementById(`${selectedCategory}`);
+        let prices = selectedDiv.getElementsByClassName("priceNum");
+        for (let i = 0; i < prices.length; i++) {
+            let currPrice = prices[i].innerHTML;
+            prices[i].innerHTML = (currPrice * (1 - selectedDiscount)).toFixed(2);
+        };
+        discountApplied[selectedSeason] = true;
+    }
+};
+
+function getDiscount(season) { //assists DiscountPrices
+    let discount;
+    seasonalSales.forEach(function(obj) {
+        if (obj.season_discount === season) {
+            discount = obj.discount;
+        }
+    });
+    return discount;
+}
+
+function getCategory(season) { //assists DiscountPrices
+    let category;
+    seasonalSales.forEach(function(obj) {
+        if (obj.season_discount === season) {
+            category = obj.name;
+        }
+    });
+    return category;
+}
+
+function getProducts() { //parses JSON to JS
+    let json = JSON.parse(this.responseText);
+    products = json.products;
+    printProducts(); //triggers printing to DOM
+}
+
+function getSeasonalSales() { //parses JSON to JS
+    let json = JSON.parse(this.responseText);
+    seasonalSales = json.categories;
+}
+
+function printProducts() { //prints to the DOM
+    let apparelString = "";
```

Comment:
Is there a better reason to rewrite this code? It's not pretty but it gets the job done. To rewrite it, I would have to build in logic to create a different div for each category, then push all the strings into the appropriate div using my existing if/else statement, then close each div out. In fact I think it might require even more code. It's easier just to have those divs exist on the HTML page. Sorry for the pushback, but if my code works... it works. For such a simple project, the method I used is not terribly inefficient.